crates/runtime/src/traphandlers.rs

```diff
@@ -457,6 +466,13 @@ impl<'a> CallThreadState<'a> {
                     maybe_interrupted,
                 })
             }
+            UnwindReason::Exit { status } => {
+                if let Some(status) = NonZeroI32::new(status) {
+                    Err(Trap::Exit { status })
+                } else {
+                    Ok(())
```

Comment:
This feels like it's somewhat dangerous here to me, should this be a panic or should `status` already have the type `NonZeroI32`?